libs/database/src/lib/entities/polygon-geometry.entity.ts

```diff
@@ -22,4 +23,88 @@ export class PolygonGeometry extends Model<PolygonGeometry> {
   @AllowNull
   @Column(BIGINT.UNSIGNED)
   createdBy: number | null;
+
+  static async checkIsSimple(polygonUuid: string): Promise<boolean> {
+    if (this.sequelize == null) {
+      throw new InternalServerErrorException("PolygonGeometry model is missing sequelize connection");
+    }
+
+    const result = (await this.sequelize.query(
+      `
+        SELECT ST_IsSimple(geom) as isSimple
+        FROM polygon_geometry 
+        WHERE uuid = :polygonUuid
+      `,
+      {
+        replacements: { polygonUuid },
+        type: QueryTypes.SELECT
+      }
+    )) as { isSimple: boolean }[];
+
+    if (result.length === 0) {
+      throw new NotFoundException();
+    }
+
+    return result[0].isSimple;
+  }
+
+  static async checkIsSimpleBatch(polygonUuids: string[]): Promise<{ uuid: string; isSimple: boolean }[]> {
+    if (this.sequelize == null) {
+      throw new InternalServerErrorException("PolygonGeometry model is missing sequelize connection");
+    }
+
+    return (await this.sequelize.query(
+      `
+        SELECT uuid, ST_IsSimple(geom) as isSimple
+        FROM polygon_geometry 
+        WHERE uuid IN (:polygonUuids)
+      `,
+      {
+        replacements: { polygonUuids },
+        type: QueryTypes.SELECT
+      }
+    )) as { uuid: string; isSimple: boolean }[];
+  }
+
+  static async getGeoJSON(polygonUuid: string): Promise<string> {
+    if (this.sequelize == null) {
+      throw new InternalServerErrorException("PolygonGeometry model is missing sequelize connection");
+    }
+
+    const result = (await this.sequelize.query(
+      `
+        SELECT ST_AsGeoJSON(geom) as geoJson
+        FROM polygon_geometry 
+        WHERE uuid = :polygonUuid
+      `,
+      {
+        replacements: { polygonUuid },
+        type: QueryTypes.SELECT
+      }
+    )) as { geoJson: string }[];
+
+    if (result.length === 0) {
+      throw new NotFoundException();
+    }
```

Comment:
Same thing here, simply return `result[0]?.geoJson` and let the validator decide whether or not to throw the error.